axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceTermServiceImpl.java

```diff
@@ -442,4 +444,116 @@ public void refusalToPay(
 
     invoiceTermRepo.save(invoiceTerm);
   }
+
+  @Override
+  @Transactional
+  public void retrieveEligibleTerms(PaymentSession paymentSession) {
+    List<InvoiceTerm> eligibleInvoiceTermList =
+        invoiceTermRepo
+            .all()
+            .filter(
+                "self.moveLine.move.company = :company AND self.paymentMode = :paymentMode AND self.dueDate <= :paymentDatePlusMargin AND self.dueDate >= :paymentDateMinusMargin AND (self.invoice.currency = :currency OR self.moveLine.move.currency = :currency) AND self.bankDetails != NULL"
+                    + " AND (self.moveLine != NULL AND (self.moveLine.move.partner.isCustomer = TRUE OR self.moveLine.move.partner.isSupplier = TRUE OR self.moveLine.move.partner.isEmployee = TRUE) OR self.moveLine = NULL AND (self.invoice.partner.isCustomer = TRUE OR self.invoice.partner.isSupplier = TRUE OR self.invoice.partner.isEmployee = TRUE))"
+                    + " AND ((self.moveLine != NULL AND self.moveLine.account.isRetrievedOnPaymentSession = TRUE) OR (self.moveLine = NULL AND self.invoice.partnerAccount.isRetrievedOnPaymentSession = TRUE))"
+                    + " AND (self.invoice.company.accountConfig.isManagePassedForPayment = FALSE OR ((self.invoice.company.accountConfig.isManagePassedForPayment = TRUE OR self.moveLine.move.company.accountConfig.isManagePassedForPayment = TRUE) AND (self.pfpValidateStatusSelect = 2 OR self.pfpValidateStatusSelect = 4)))"
+                    + " AND self.isPaid = FALSE AND self.amountRemaining > 0 AND self.paymentSession = NULL")
```

Comment:
`IS NULL`/`IS NOT NULL` must be used instead of `= NULL`/`!= NULL` here.
The full query should also be rewritten to be more understandable:
- At least, more line break must be used (maximum line length in our java code is 100 chars and we have a line with 300 chars)
- Then, maybe break down the query in multiple constants. (for example, one constant per line in the current query would allow to read more easily each part)